### PR TITLE
Add GetEntityIfExists method to TableClient

### PR DESCRIPTION
**Pull request by christothes**
related to #16251
related to #25626

### CODE REVIEW

File: sdk/tables/Azure.Data.Tables/src/TableClient.cs [hunk 1]
```diff
@@ -513,14 +515,17 @@ public virtual async Task<Response<TableItem>> CreateIfNotExistsAsync(Cancellati
...
+                    new TablesRequestContext(TablesRequestContext.Conflict) { CancellationToken = cancellationToken, ErrorOptions = ErrorOptions.NoThrow };
+
                 var response = await _tableOperations.CreateAsync(
                     RequestContent.Create(new TableProperties { TableName = Name }),
                     TableConstants.Odata.MinimalMetadata,
                     TableConstants.ReturnNoContent,
                     context).ConfigureAwait(false);
 
-                if (response.IsError)
+                if (response.IsError || response.Status == (int)HttpStatusCode.Conflict)
```
Comment (jsquire) [inline sdk/tables/Azure.Data.Tables/src/TableClient.cs:528]: Should we consider using the `public static` member from the request context here and avoid the cast?

File: sdk/tables/Azure.Data.Tables/src/TableClient.cs [hunk 2]
```diff
@@ -676,12 +681,49 @@ public virtual Response<T> GetEntity<T>(string partitionKey, string rowKey, IEnu
...
+        /// <param name="partitionKey">The partitionKey that identifies the table entity.</param>
+        /// <param name="rowKey">The rowKey that identifies the table entity.</param>
+        /// <param name="select">Selects which set of entity properties to return in the result set. Pass <c>null</c> to retreive all properties.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns>The <see cref="Response"/> indicating the result of the operation.</returns>
+        /// <exception cref="RequestFailedException">Exception thrown if the entity doesn't exist.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="partitionKey"/> or <paramref name="rowKey"/> is null.</exception>
+        // Supression is required to avoid an ambiguous call site
+#pragma warning disable AZC0002 // DO ensure all service methods, both asynchronous and synchronous, take an optional CancellationToken parameter called cancellationToken.
```
Comment (jsquire) [inline sdk/tables/Azure.Data.Tables/src/TableClient.cs:698]: This is a weird disable since you have the parameter in here...

**Review by jsquire — APPROVED** [review]

File: sdk/tables/Azure.Data.Tables/src/TableClient.cs (hunk 2 above)
Comment (christothes) [reply sdk/tables/Azure.Data.Tables/src/TableClient.cs:698]: oh this is left over from earlier experiments - thanks for catching!

File: sdk/tables/Azure.Data.Tables/src/TableClient.cs (hunk 1 above)
Comment (christothes) [reply sdk/tables/Azure.Data.Tables/src/TableClient.cs:528]: If you are referring to the static on TablesRequestContext, that one is an `int[] { (int)HttoStatusCode.Conflict }` - I suppose we could index into that array, but I don't think the cast is very expensive.

Comment (jsquire) [reply sdk/tables/Azure.Data.Tables/src/TableClient.cs:528]: ahh... yeah.  My bad, that's what I was thinking of but forgot it was an array.   No concerns on the cast, just thought we had an easily available alias for it already.

**Pull request merged**